llvm/test/CodeGen/AMDGPU/copy-hoist-no-spills.ll

```diff
@@ -0,0 +1,94 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 4
```

Comment:
The check lines do not look auto-generated, but they might be a bit fragile as-is. Would it be possible to auto-generate them? Or is that too large?
An explanation of what is going wrong in the test (some-such thing should not be hoisted and spilled?) might be useful in the long-run too.

Reply:
I started with autogenerated test, hence the artefact in the NOTE.
But I liked my way of checking much more than the autogeneration. The latter is going to fail from the every little sneeze in the compiler that changes sequence of instructions of register names. Instead my way of checking checks exactly what I want: spill reload in the loop. And when the test fails the error is very clear to read.